packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js

```diff
@@ -155,6 +145,61 @@ export default class {
     }
   }
 
+  setClusterRegion(isRegionEnabled = false) {
+    this.cluster.region = new Datacenter({
+      name: this.cluster.region.name,
+      enabled: isRegionEnabled || this.cluster.region.enabled,
+      quota: find(this.quotas, { region: this.cluster.region.name }),
+    });
+  }
+
+  onRegionSubmit() {
+    this.setClusterRegion();
+    this.loadPrivateNetworks();
+
+    if (this.cluster.region.enabled) {
+      this.loadFlavors(this.cluster.region.name);
+    }
+
+    this.displaySelectedRegion = true;
+  }
+
+  onKubeVersionFocus() {
+    const { region } = this.cluster;
+    const { name, enabled } = region;
+    this.displaySelectedRegion = true;
+
+    if (!enabled) {
+      this.isAddingNewRegion = true;
+
+      return this.Kubernetes.addRegion(this.projectId, region)
```

Comment:
no need to show the loader on UI when region is being added?

Reply:
Yes it's already here:
`this.isAddingNewRegion = true;`